nova/exception.py

```diff
@@ -1244,6 +1244,9 @@ class FlavorMemoryTooSmall(NovaException):
 class FlavorDiskTooSmall(NovaException):
     msg_fmt = _("Flavor's disk is too small for requested image.")
 
+class BlockDeviceSizeTooSmall(NovaException):
+    msg_fmt = _("Root Block device is too small for requested image.%(reason)s")
```

Comment:
Can you check whether the intended reason is being displayed in place of %(reason)s ?